### PR TITLE
feat: Optimize trace chain elements

### PR DESCRIPTION
**Pull request by denis-tingaikin**
## Description

Logging for 
- FATAL, ERROR, WARN: work 100 times faster.
- TRACE, DEBUG: work 7 times faster.
- INFO: difference at the margin of error.

## Issue link
https://github.com/networkservicemesh/sdk/issues/1667

## How Has This Been Tested?

- [X] Added unit testing to cover
- [X] Tested manually
- [X] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [ ] Bug fix
- [ ] New functionality
- [ ] Documentation
- [X] Refactoring
- [ ] CI

### CODE REVIEW

**Comment by denis-tingaikin** [comment]
We're aiming to get this PR in RC1. Feel free to comment; we’ll merge it into main in one to two weeks after testing.

File: pkg/networkservice/core/trace/server_bench_test.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	res.GetConnection().Id = res.GetConnection().GetPath().GetPathSegments()[0].Id
+
+	return res
+}
+
+func newTestServerChain(size int) networkservice.NetworkServiceServer {
+	var servers []networkservice.NetworkServiceServer
+
+	for i := 0; i < size; i++ {
+		id := i
```
Comment (NikitaSkrynnik) [inline pkg/networkservice/core/trace/server_bench_test.go:61]: As far as I remember in go 1.23 we don't need to declare local vars for loop indices

File: pkg/networkservice/core/trace/server_bench_test.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+	log.EnableTracing(true)
+
+	b.ResetTimer()
+
+	for range b.N {
+		_, _ = s.Request(ctx, request)
+	}
+}
+
+func Benchmark_LongRequest_Diff_Warn(b *testing.B) {
```
Comment (NikitaSkrynnik) [inline pkg/networkservice/core/trace/server_bench_test.go:159]: Is it a proper name for this test? Shouldn't it be `Benchmark_LongRequest_NoDiff_Warn`?

File: pkg/tools/stringutils/stringutils.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+// limitations under the License.
+
+// Package stringutils provides helper functions related to strings.
+package stringutils
+
+import "unsafe"
+
+// CovertBytesToString converts slice of bytes to string via unsafe package with zero allocations.
+// Note: slice MUST NOT be modified after using this function.
+func CovertBytesToString(slice []byte) string {
```
Comment (NikitaSkrynnik) [inline pkg/tools/stringutils/stringutils.go:24]:
```suggestion
func ConvertBytesToString(slice []byte) string {
```

File: pkg/networkservice/core/trace/traceverbose/common.go [hunk 4]
```diff
@@ -216,14 +206,19 @@ func logError(ctx context.Context, err error, operation string) error {
...
 	msg := ""
 	cc, err := json.Marshal(v)
 	if err == nil {
-		msg = string(cc)
+		msg = stringutils.CovertBytesToString(cc)
 	} else {
 		msg = fmt.Sprint(v)
 	}
-	s.Tracef("%v=%s", k, msg)
+	s.Infof("%v=%s", k, msg)
```
Comment (NikitaSkrynnik) [inline pkg/networkservice/core/trace/traceverbose/common.go:223]: Why do we change this?

File: pkg/networkservice/core/trace/server_bench_test.go (hunk 1 above)
Comment (denis-tingaikin) [reply pkg/networkservice/core/trace/server_bench_test.go:61]: Thanks, fixed.

File: pkg/networkservice/core/trace/server_bench_test.go (hunk 2 above)
Comment (denis-tingaikin) [reply pkg/networkservice/core/trace/server_bench_test.go:159]: Thanks, fixed.

File: pkg/networkservice/core/trace/traceverbose/common.go (hunk 4 above)
Comment (denis-tingaikin) [reply pkg/networkservice/core/trace/traceverbose/common.go]: Thanks, fixed.

File: pkg/tools/stringutils/stringutils.go (hunk 3 above)
Comment (denis-tingaikin) [reply pkg/tools/stringutils/stringutils.go]: Thanks, fixed.

**Pull request merged**